### PR TITLE
onl: clean up build and reduce amount of variables needed

### PR DESCRIPTION
**Pull request by KanjiMonster**
Make use of the ONL build system to autodetect build paths, which allows
us to reduce the amount of variables significantly, basically reducing the
need to define any ONL_* variables by machines.

For each platform in ONL there exists a package named onlp-<platform>.
Using the ONL package manager (onlpm), we can find out where in the
filesystem this package is located, and do not need to hardcode the path
to it anymore.

Sideeffect is a slight increase in build time, as it needs to build its
package database on the first invocation, which delays the build by a
few seconds.

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>

### CODE REVIEW

**Comment by rubensfig** [comment]
Build fails with the following error on the 4630 platform.

```
make: Leaving directory '/tmp/accton-as4630-54pe/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/base/any/onlp/builds/onlpd'
Traceback (most recent call last):
File "tools/onlpm.py", line 23, in <module>
       import lsb_release
       ImportError: No module named lsb_release
WARNING: exit code 1 from a shell command.
ERROR: Execution of '/tmp/accton-as4630-54pe/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/temp/run.do_compile.23331' failed with exit code 1
```

**Comment by KanjiMonster** [comment]
This python2 thing is getting annoying. Let's fix it. Will rebase onto #52, then this won't be an issue anymore.

**Comment by KanjiMonster** [comment]
Updated and rebased on master with the switch to python3:

* fixed some leftovers in the python3 conversion patch
* removed the need for the lsb_release python module
* dropped BISDN_ONIE_PLATFORM

**Review by rubensfig — APPROVED** [review]

**Review by ideaship — CHANGES_REQUESTED** [review]

File: recipes-extended/onl/files/onl/0008-onlpm-add-an-argument-for-just-printing-the-builddir.patch [hunk 1]
```diff
@@ -0,0 +1,49 @@
+Upstream-Status: Inappropriate [Hack for building with yocto]
+
+From ae45c22f92901b54ae817fd8bcf2f5bae3d5c205 Mon Sep 17 00:00:00 2001
+From: Jonas Gorski <jonas.gorski@bisdn.de>
+Date: Wed, 18 May 2022 15:49:13 +0200
+Subject: [PATCH 2/3] onlpm: add an argument for just printing the builddirs
+
+To allow building without packaging, we need to know the appropriate
+build dir execute make in.
```
[the remark this answers is not in the document]
Comment (KanjiMonster) [reply recipes-extended/onl/files/onl/0008-onlpm-add-an-argument-for-just-printing-the-builddir.patch]: Reworded slightly.

**Review by ideaship — APPROVED** [review]
LGTM

**Pull request merged**